src/julia-parser.scm

```diff
@@ -1341,7 +1341,7 @@
         (loop (cons (symbol (string.sub (string nxt) 1))
                     path)))
        (else
-        (error (string "invalid \"" word "\" statement")))))))
+        `(,word ,@(reverse path)))))))
```

Comment:
IIRC, the logic here is similar to the one used for `export` and if I understand correctly, that case is rejected by the surrounding logic. (the parser for `import` and `export` just leave the unused token in the stream and the surrounding logic will raise an error if what's left is not a proper place to end a statement)

On my branch.

```julia
julia> using Base 1
ERROR: syntax: extra token "1" after end of expression

julia> using Base (
ERROR: syntax: extra token "(" after end of expression

julia> using Base ()
ERROR: syntax: extra token "(" after end of expression

julia> :(using Base ())
ERROR: syntax: missing separator in tuple

julia> :(using Base A)
ERROR: syntax: missing separator in tuple

julia> :(using Base)
:(using Base)

julia> :(using Base)
:(using Base)

julia> :(using Base 1)
ERROR: syntax: missing separator in tuple
```